rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/validator/ValidatorGenerator.xtend

```diff
@@ -0,0 +1,202 @@
+import com.regnosys.rosetta.rosetta.simple.Attribute
+import com.regnosys.rosetta.rosetta.simple.Data
+import com.regnosys.rosetta.types.RDataType
+import com.regnosys.rosetta.types.RosettaTypeProvider
+import com.regnosys.rosetta.types.TypeSystem
+import com.regnosys.rosetta.types.builtin.RBuiltinTypeService
+import com.regnosys.rosetta.types.builtin.RNumberType
+import com.regnosys.rosetta.types.builtin.RStringType
+import com.rosetta.model.lib.ModelSymbolId
+import com.rosetta.model.lib.validation.AttributeValidation
+import com.rosetta.model.lib.validation.RosettaModelObjectValidator
+import com.rosetta.model.lib.validation.TypeValidation
+import com.rosetta.model.lib.validation.ValidationUtil
+import com.rosetta.util.DottedPath
+import java.math.BigDecimal
+import java.util.ArrayList
+import java.util.Optional
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+import javax.inject.Inject
+import org.apache.commons.text.StringEscapeUtils
+import org.eclipse.xtend2.lib.StringConcatenationClient
+import org.eclipse.xtext.generator.IFileSystemAccess2
+import static extension com.regnosys.rosetta.generator.util.RosettaAttributeExtensions.*
+import java.util.List
+import com.rosetta.model.lib.validation.ValidationResult
+import com.rosetta.model.lib.path.RosettaPath
+import com.rosetta.model.lib.validation.ConditionValidation
+import com.rosetta.util.types.generated.GeneratedJavaClass
+import java.util.Set
+import java.util.Map
+import com.rosetta.model.lib.validation.ExistenceChecker
+import com.rosetta.model.lib.validation.ValidatorWithArg
+import com.google.common.collect.ImmutableMap
+import com.rosetta.model.lib.validation.ValidationData
+import java.util.concurrent.ConcurrentHashMap
+import com.regnosys.rosetta.types.RObjectFactory
+
+class ValidatorGenerator {
+	@Inject extension ImportManagerExtension
+	@Inject extension RosettaExtensions
+	@Inject extension JavaTypeTranslator
+	@Inject extension RosettaTypeProvider
+	@Inject extension TypeSystem
+	@Inject extension RBuiltinTypeService
+	@Inject extension JavaTypeUtil
+	@Inject extension RObjectFactory
+
+	def generate(RootPackage root, IFileSystemAccess2 fsa, Data data, String version) {
+		val topScope = new JavaScope(root.typeValidation)
+
+		val classBody = data.classBody(topScope, root)
+		val content = buildClass(root.typeValidation, classBody, topScope)
+		fsa.generateFile('''«root.typeValidation.withForwardSlashes»/«data.name»Validator.java''', content)
+	}
+
+	private def StringConcatenationClient classBody(Data data, JavaScope scope, RootPackage root) {
+
+		val modelPojo = new RDataType(data).toJavaReferenceType
+		val rDataType = new RDataType(data)
+		'''
+			public class «data.name»Validator implements «RosettaModelObjectValidator»<«modelPojo»>{
+				«FOR con : data.conditions»
+				@«Inject» protected «new GeneratedJavaClass(root.condition, con.conditionName(data).toConditionJavaType, Object)» «con.conditionName(data).toFirstLower» ;
```

Comment:
About the name `«con.conditionName(data).toFirstLower»` - this may fail to generate valid Java code in certain edge cases, which I haven't explained to you yet. Remind me to explain usage of the `JavaScope` class to you. :)